src/state.rs

```diff
@@ -64,6 +64,14 @@ impl NetworkState {
         }
     }
 
+    /// Mutates the `block_height` attr of the current `NetworkState` instance.
+    ///
+    /// That method is temporary, and will not be mantained in the API on the
+    /// next versions.
+    pub fn set_new_block_height(&mut self, block_height: u64) {
```

Comment:
Remove the `new`, it's just a setter (so `set_block_height`).

If you want to keep the `new` it means this is not a regular setter, you could for example performs a check inside to ensure the new block height is actually greater than the current, as guard / safety check. However in that case you should returns a `boolean` to indicate if the `block_height` was actually mutated.